File: exercism/javascript/hamming/hamming.js
// complete on May 9, 2017


Hamming = function() {};

counter = function(str1, str2) {
  var diff = 0;
  for (var i = 0; i <= str1.length - 1; i++)
  { 
    if (str1[i] !== str2[i]) {
      diff ++;
    }
  }
  return diff;
};

Hamming.prototype.compute = function(dna1, dna2) {
  if (dna1.length !== dna2.length) throw new Error('DNA strands must be of equal length.');
  return counter(dna1, dna2);
};

module.exports = Hamming;

File: exercism/python/hamming/hamming.py
#Complete Aug 8, 2017

def distance(dna1, dna2):
    if (len(dna1) != len(dna2)):
        raise ValueError
    else:
        length = len(dna1)
        distance = 0
        for i in xrange(length):
            if dna1[i] != dna2[i]:
                distance += 1
        return distance
